build-logic/convention/build.gradle.kts

```diff
@@ -35,9 +34,9 @@ gradlePlugin {
       id = "io.getstream.video.android.library"
       implementationClass = "AndroidLibraryConventionPlugin"
     }
-    register("spotless") {
-      id = "io.getstream.spotless"
-      implementationClass = "SpotlessConventionPlugin"
+    register("demoFlavorConvention") {
+      id = "io.getstream.video.android.demoflavor"
+      implementationClass = "DemoFlavorConventionPlugin"
```

Comment:
When removing the Spotless plugin, `demo-app` stopped having a dependency on `build-logic` and started failing because it couldn't resolve `configureFlavors`. So I moved that logic to this small "demo flavor" plugin